src/components/number-input/number-input.ts

```diff
@@ -70,7 +70,7 @@ export default class BXNumberInput extends BXInput {
   }
 
   _getValidityMessage(state: string) {
-    if (Object.keys(NUMBER_INPUT_VALIDATION_STATUS).includes(state)) {
+    if (Object.values(NUMBER_INPUT_VALIDATION_STATUS).includes(state)) {
```

Comment:
```suggestion
    if (Object.values(NUMBER_INPUT_VALIDATION_STATUS).includes(state as NUMBER_INPUT_VALIDATION_STATUS)) {
```

This may solve the TS check error... Sorry if it's not the case.